sdks/java/core/src/test/java/org/apache/beam/sdk/io/WriteFilesTest.java

```diff
@@ -391,11 +455,17 @@ private void runShardedWrite(
       timestamps.add(i + 1);
     }
 
-    SimpleSink sink = makeSimpleSink();
+    SimpleSink sink = makeSimpleSink(windowedWrites);
```

Comment:
I think it will be easier to read all the tests if, instead of passing the parameters such as `numConfiguredShards`, `windowedWrites` etc., you make `runShardedWrite` take a `WriteFiles` instance directly.

It's easier to read `WriteFiles.to(makeSimpleSink(true) /* windowedWrites */).withMaxNumWritersPerBundle(42)` than "..., true, Optional.of(42)".